packages/fiori/src/SideNavigation.ts

```diff
@@ -51,20 +52,22 @@ type SideNavigationPopoverContents = {
 };
 
 type SideNavigationSelectionChangeEventDetail = {
-	item: SideNavigationItemBase;
+	item: SideNavigationItemBase,
 };
 
 // used for the inner side navigation used in the SideNavigationPopoverTemplate
 type PopupClickEventDetail = {
 	target: {
-		associatedItem: SideNavigationItemBase
+		associatedItem: SideNavigationItemBase,
 	}
 };
 
 // used for the inner side navigation used in the SideNavigationPopoverTemplate
-type NavigationMenuClickEventDetail = {
-	item: {
-		associatedItem: SideNavigationItemBase
+type NavigationMenuClickEventDetail = MenuItemClickEventDetail & {
```

Comment:
```ts
type NavigationMenuClickEventDetail = MenuItemClickEventDetail & {
	item: Pick<MenuItemClickEventDetail, "item"> & {
		associatedItem: SideNavigationItemBase,
	}
};
```

It will take type of `item` defined in MenuItemClickEventDetail and it will enhance it will associatedItem property